server/services/offendersService.js

```diff
@@ -742,12 +762,15 @@ module.exports = function createOffendersService(
           return null
         }
 
-        const releaseDate = releaseDateMap.get(raw.bookingId)
+        const releaseDateForDecidingIfRecordShouldBeIncluded =
+          !raw.dbRecord || !raw.dbRecord.catType || raw.dbRecord.catType === CatType.RECAT.name
```

Comment:
This is to match the logic on line 57, since a different function is now being called to get the release date (getprisonerSearchData).